--- neutrino-mp-tangos.org/src/neutrino.cpp
+++ neutrino-mp-tangos/src/neutrino.cpp
@@ -4149,6 +4149,9 @@
 			CVFD::getInstance()->Clear();
 			CVFD::getInstance()->setMode(CVFD::MODE_STANDBY);
 		}
+#if defined (FORTIS_HDBOX) || defined (TF7700)
+		CVFD::getInstance()->setlcdparameter(g_settings.lcd_setting[SNeutrinoSettings::LCD_STANDBY_BRIGHTNESS], 0);
+#endif
 		CVFD::getInstance()->setBacklight(g_settings.backlight_standby);
 
 		InfoClock->enableInfoClock(false);
@@ -4180,6 +4183,9 @@
 		// Active standby off
 		powerManager->SetStandby(false, false);
 		CVFD::getInstance()->setMode(CVFD::MODE_TVRADIO);
+#if defined (FORTIS_HDBOX) || defined (TF7700)
+		CVFD::getInstance()->setlcdparameter(g_settings.lcd_setting[SNeutrinoSettings::LCD_BRIGHTNESS], 1);
+#endif
 		CVFD::getInstance()->ShowText("resume");
 		cpuFreq->SetCpuFreq(g_settings.cpufreq * 1000 * 1000);
 		videoDecoder->Standby(false);
--- neutrino-mp-tangos/src/driver/simple_display.cpp.org
+++ neutrino-mp-tangos/src/driver/simple_display.cpp
@@ -35,6 +35,9 @@
 //#include <math.h>
 #include <sys/stat.h>
 #if HAVE_SPARK_HARDWARE
+struct semaphore {
+	int dummy;
+};
 #include <aotom_main.h>
 #define DISPLAY_DEV "/dev/vfd"
 #endif
--- neutrino-mp-tangos/data/icons/Makefile.am.org
+++ neutrino-mp-tangos/data/icons/Makefile.am
@@ -263,7 +263,7 @@
 	shell.png \
 	shutdown.jpg \
 	softupdate.png \
-	star-off \
+	star-off.png \
 	star-on.png \
 	start.jpg \
 	streaming.png \

