--- enigma2-nightly.org/lib/python/Tools/KeyBindings.py
+++ enigma2-nightly/lib/python/Tools/KeyBindings.py
@@ -5,7 +5,7 @@
 from Components.config import config
 from Components.RcModel import rc_model
 
-keyDescriptions = [{
+keyDescriptions = [{ # for remotes with ID 0 (DMM)
 		KEYIDS["BTN_0"]: ("UP", "fp"),
 		KEYIDS["BTN_1"]: ("DOWN", "fp"),
 		KEYIDS["KEY_OK"]: ("OK", ""),
@@ -47,7 +47,7 @@
 		KEYIDS["KEY_STOP"]: ("STOP",),
 		KEYIDS["KEY_RECORD"]: ("RECORD",)
 	},
-	{
+	{  # for remotes with ID 1 (DMM advanced)
 		KEYIDS["BTN_0"]: ("UP", "fp"),
 		KEYIDS["BTN_1"]: ("DOWN", "fp"),
 		KEYIDS["KEY_OK"]: ("OK", ""),
@@ -87,9 +87,12 @@
 		KEYIDS["KEY_9"]: ("9",),
 		KEYIDS["KEY_EXIT"]: ("EXIT",),
 		KEYIDS["KEY_STOP"]: ("TV", "SHIFT"),
-		KEYIDS["KEY_RECORD"]: ("RADIO", "SHIFT")
-	},
-	{
+		KEYIDS["KEY_RECORD"]: ("RADIO", "SHIFT"),
+		KEYIDS["KEY_EPG"]: ("EPG",),
+		KEYIDS["KEY_REWIND"]: ("REWIND",),
+		KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",)
+	},
+	{ # for remotes with ID 2 (default)
 		KEYIDS["BTN_0"]: ("UP", "fp"),
 		KEYIDS["BTN_1"]: ("DOWN", "fp"),
 		KEYIDS["KEY_OK"]: ("OK", ""),
@@ -131,36 +134,12 @@
 		KEYIDS["KEY_STOP"]: ("STOP",),
 		KEYIDS["KEY_RECORD"]: ("RECORD",)
 	},
-	{       # XP1000
-		KEYIDS["BTN_0"]: ("UP", "fp"),
-		KEYIDS["BTN_1"]: ("DOWN", "fp"),
-		KEYIDS["KEY_OK"]: ("OK", ""),
-		KEYIDS["KEY_UP"]: ("UP",),
-		KEYIDS["KEY_DOWN"]: ("DOWN",),
+	{  # for remotes with ID 3 (Spark RC12)
 		KEYIDS["KEY_POWER"]: ("POWER",),
-		KEYIDS["KEY_RED"]: ("RED",),
-		KEYIDS["KEY_BLUE"]: ("BLUE",),
-		KEYIDS["KEY_GREEN"]: ("GREEN",),
-		KEYIDS["KEY_YELLOW"]: ("YELLOW",),
-		KEYIDS["KEY_MENU"]: ("MENU",),
-		KEYIDS["KEY_LEFT"]: ("LEFT",),
-		KEYIDS["KEY_RIGHT"]: ("RIGHT",),
-		KEYIDS["KEY_VIDEO"]: ("PVR",),
-		KEYIDS["KEY_INFO"]: ("INFO",),
-		KEYIDS["KEY_AUDIO"]: ("AUDIO",),
-		KEYIDS["KEY_SUBTITLE"]: ("SUBTITLE",),
-		KEYIDS["KEY_TV"]: ("TV",),
-		KEYIDS["KEY_RADIO"]: ("RADIO",),
-		KEYIDS["KEY_TEXT"]: ("TEXT",),
-		KEYIDS["KEY_NEXT"]: ("ARROWRIGHT",),
-		KEYIDS["KEY_PREVIOUS"]: ("ARROWLEFT",),
-		KEYIDS["KEY_PREVIOUSSONG"]: ("SKIPBACK",),
-		KEYIDS["KEY_REWIND"]: ("REWIND",),
-		KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",),
-		KEYIDS["KEY_NEXTSONG"]: ("SKIPFORWARD",),
-		KEYIDS["KEY_PLAYPAUSE"]: ("PLAYPAUSE",),
-		KEYIDS["KEY_CHANNELUP"]: ("BOUQUET+",),
-		KEYIDS["KEY_CHANNELDOWN"]: ("BOUQUET-",),
+		KEYIDS["KEY_V"]: ("V.FORMAT",),
+		KEYIDS["KEY_AUX"]: ("TV/SAT",),
+		KEYIDS["KEY_TIME"]: ("TIME",),
+		KEYIDS["KEY_MUTE"]: ("MUTE",),
 		KEYIDS["KEY_0"]: ("0",),
 		KEYIDS["KEY_1"]: ("1",),
 		KEYIDS["KEY_2"]: ("2",),
@@ -171,63 +150,204 @@
 		KEYIDS["KEY_7"]: ("7",),
 		KEYIDS["KEY_8"]: ("8",),
 		KEYIDS["KEY_9"]: ("9",),
-		KEYIDS["KEY_EXIT"]: ("EXIT",),
-		KEYIDS["KEY_STOP"]: ("STOP",),
-		KEYIDS["KEY_RECORD"]: ("RECORD",),
-		KEYIDS["KEY_BOOKMARKS"]: ("PORTAL",),
-		KEYIDS["KEY_VMODE"]: ("VMODE",),
-		KEYIDS["KEY_PROGRAM"]: ("TIMER",),
-		KEYIDS["KEY_SLEEP"]: ("SLEEP",),
-		KEYIDS["KEY_EPG"]: ("EPG",),
-	},
-	{       # Formuler F1
-		KEYIDS["BTN_0"]: ("UP", "fp"),
-		KEYIDS["BTN_1"]: ("DOWN", "fp"),
-		KEYIDS["KEY_OK"]: ("OK", ""),
+		KEYIDS["KEY_TV2"]: ("TV/RADIO",),
+		KEYIDS["KEY_BACK"]: ("RECALL",),
+		KEYIDS["KEY_PAGEUP"]: ("PAGE+",),
+		KEYIDS["KEY_PAGEDOWN"]: ("PAGE-",),
+		KEYIDS["KEY_FIND"]: ("FIND",),
+		KEYIDS["KEY_ARCHIVE"]: ("FOLDER",),
+		KEYIDS["KEY_VOLUMEUP"]: ("VOL+",),
+		KEYIDS["KEY_VOLUMEDOWN"]: ("VOL-",),
+		KEYIDS["KEY_MENU"]: ("MENU",),
+		KEYIDS["KEY_INFO"]: ("INFO",),
 		KEYIDS["KEY_UP"]: ("UP",),
 		KEYIDS["KEY_DOWN"]: ("DOWN",),
-		KEYIDS["KEY_POWER"]: ("POWER",),
+		KEYIDS["KEY_LEFT"]: ("LEFT",),
+		KEYIDS["KEY_RIGHT"]: ("RIGHT",),
+		KEYIDS["KEY_OK"]: ("OK", ""),
+		KEYIDS["KEY_HOME"]: ("EXIT",),
+		KEYIDS["KEY_EPG"]: ("EPG",),
+		KEYIDS["KEY_FAVORITES"]: ("FAV",),
+		KEYIDS["KEY_SAT"]: ("SAT",),
 		KEYIDS["KEY_RED"]: ("RED",),
-		KEYIDS["KEY_BLUE"]: ("BLUE",),
 		KEYIDS["KEY_GREEN"]: ("GREEN",),
 		KEYIDS["KEY_YELLOW"]: ("YELLOW",),
-		KEYIDS["KEY_MENU"]: ("MENU",),
-		KEYIDS["KEY_LEFT"]: ("LEFT",),
-		KEYIDS["KEY_RIGHT"]: ("RIGHT",),
-		KEYIDS["KEY_VIDEO"]: ("PVR",),
-		KEYIDS["KEY_INFO"]: ("INFO",),
-		KEYIDS["KEY_AUDIO"]: ("AUDIO",),
-		KEYIDS["KEY_TV"]: ("TV",),
-		KEYIDS["KEY_RADIO"]: ("RADIO",),
-		KEYIDS["KEY_TEXT"]: ("TEXT",),
+		KEYIDS["KEY_BLUE"]: ("BLUE",),
+		KEYIDS["KEY_RECORD"]: ("RECORD",),
+		KEYIDS["KEY_STOP"]: ("STOP",),
+		KEYIDS["KEY_PLAY"]: ("PLAY",),
+		KEYIDS["KEY_PAUSE"]: ("PAUSE",),
+		KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",),
+		KEYIDS["KEY_REWIND"]: ("REWIND",),
+		KEYIDS["KEY_PREVIOUS"]: ("ARROWLEFT",),
 		KEYIDS["KEY_NEXT"]: ("ARROWRIGHT",),
-		KEYIDS["KEY_PREVIOUS"]: ("ARROWLEFT",),
-		KEYIDS["KEY_REWIND"]: ("REWIND",),
-		KEYIDS["KEY_PAUSE"]: ("PAUSE",),
-		KEYIDS["KEY_PLAY"]: ("PLAY",),
-		KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",),
-		KEYIDS["KEY_CHANNELUP"]: ("BOUQUET+",),
-		KEYIDS["KEY_CHANNELDOWN"]: ("BOUQUET-",),
-		KEYIDS["KEY_0"]: ("0",),
-		KEYIDS["KEY_1"]: ("1",),
-		KEYIDS["KEY_2"]: ("2",),
-		KEYIDS["KEY_3"]: ("3",),
-		KEYIDS["KEY_4"]: ("4",),
-		KEYIDS["KEY_5"]: ("5",),
-		KEYIDS["KEY_6"]: ("6",),
-		KEYIDS["KEY_7"]: ("7",),
-		KEYIDS["KEY_8"]: ("8",),
-		KEYIDS["KEY_9"]: ("9",),
-		KEYIDS["KEY_EXIT"]: ("EXIT",),
-		KEYIDS["KEY_STOP"]: ("STOP",),
-		KEYIDS["KEY_RECORD"]: ("RECORD",),
+		KEYIDS["KEY_F"]: ("FAST",),
+		KEYIDS["KEY_SLOW"]: ("SLOW",),
+		KEYIDS["KEY_P"]: ("PLAY_MODE",),
+		KEYIDS["KEY_CLOSE"]: ("USB",),
+		KEYIDS["KEY_T"]: ("Tms",),
 		KEYIDS["KEY_F1"]: ("F1",),
 		KEYIDS["KEY_F2"]: ("F2",),
 		KEYIDS["KEY_F3"]: ("F3",),
-		KEYIDS["KEY_BACK"]: ("RECALL",),
-		KEYIDS["KEY_CONTEXT_MENU"]: ("CONTEXT",),
-		KEYIDS["KEY_EPG"]: ("EPG",),
-		KEYIDS["KEY_BOOKMARKS"]: ("PLAYLIST",),
+		KEYIDS["KEY_OPEN"]: ("PORTAL",),
+	},
+	{  # for remotes with ID 4 (All Fortis)
+		KEYIDS["KEY_MUTE"]: ("MUTE",),
+		KEYIDS["KEY_POWER"]: ("STANDBY",),
+		KEYIDS["KEY_ANGLE"]: ("VFORMAT",),
+		KEYIDS["KEY_SCREEN"]: ("RESOLUTION",),
+		KEYIDS["KEY_1"]: ("1BUTTON",),
+		KEYIDS["KEY_2"]: ("2BUTTON",),
+		KEYIDS["KEY_3"]: ("3BUTTON",),
+		KEYIDS["KEY_4"]: ("4BUTTON",),
+		KEYIDS["KEY_5"]: ("5BUTTON",),
+		KEYIDS["KEY_6"]: ("6BUTTON",),
+		KEYIDS["KEY_7"]: ("7BUTTON",),
+		KEYIDS["KEY_8"]: ("8BUTTON",),
+		KEYIDS["KEY_9"]: ("9BUTTON",),
+		KEYIDS["KEY_HELP"]: ("INFO",),
+		KEYIDS["KEY_0"]: ("0BUTTON",),
+		KEYIDS["KEY_MEMO"]: ("RECALL",),
+		KEYIDS["KEY_VOLUMEUP"]: ("VOLUMEUP",),
+		KEYIDS["KEY_MENU"]: ("MENU",),
+		KEYIDS["KEY_PAGEUP"]: ("CHANNELUP",),
+		KEYIDS["KEY_VOLUMEDOWN"]: ("VOLUMEDOWN",),
+		KEYIDS["KEY_HOME"]: ("EXIT",),
+		KEYIDS["KEY_PAGEDOWN"]: ("CHANNELDOWN",),
+		KEYIDS["KEY_UP"]: ("UP",),
+		KEYIDS["KEY_LEFT"]: ("LEFT",),
+		KEYIDS["KEY_OK"]: ("OK",),
+		KEYIDS["KEY_RIGHT"]: ("RIGHT",),
+		KEYIDS["KEY_DOWN"]: ("DOWN",),
+		KEYIDS["KEY_TEEN"]: ("UPUP",),
+		KEYIDS["KEY_EPG"]: ("EPG",),
+		KEYIDS["KEY_TWEN"]: ("DOWNDOWN",),
+		KEYIDS["KEY_REWIND"]: ("REWIND",),
+		KEYIDS["KEY_PLAY"]: ("PLAY",),
+		KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",),
+		KEYIDS["KEY_PREVIOUS"]: ("LAST",),
+		KEYIDS["KEY_RECORD"]: ("RECORD",),
+		KEYIDS["KEY_NEXT"]: ("NEXT",),
+		KEYIDS["KEY_DIRECTORY"]: ("OPEN",),
+		KEYIDS["KEY_PAUSE"]: ("PAUSE",),
+		KEYIDS["KEY_STOP"]: ("STOP",),
+		KEYIDS["KEY_SELECT"]: ("CHECK",),
+		KEYIDS["KEY_RED"]: ("RED",),
+		KEYIDS["KEY_GREEN"]: ("GREEN",),
+		KEYIDS["KEY_YELLOW"]: ("YELLOW",),
+		KEYIDS["KEY_BLUE"]: ("BLUE",),
+		KEYIDS["KEY_OPTION"]: ("PIP",),
+		KEYIDS["KEY_GOTO"]: ("SWAP",),
+#		KEYIDS["KEY_PIPSEL"]: ("PIP_SEL",), # PIP list
+		KEYIDS["KEY_SLOW"]: ("SLEEP",),
+		KEYIDS["KEY_FAVORITES"]: ("FAV",),
+		KEYIDS["KEY_MODE"]: ("TVRADIO",),
+		KEYIDS["KEY_SUBTITLE"]: ("LIST",),
+		KEYIDS["KEY_TEXT"]: ("TEXT",),
+		KEYIDS["KEY_ZOOM"]: ("ZOOM",),
+	},
+	{  # for remotes with ID 5 (Topfield TF77X0HDPVR)
+		KEYIDS["KEY_POWER"]: ("STANDBY",),
+		KEYIDS["KEY_MUTE"]: ("MUTE",),
+		KEYIDS["KEY_ZOOM"]: ("V.FORMAT",),
+		KEYIDS["KEY_SCREEN"]: ("A/R",),
+		KEYIDS["KEY_AUX"]: ("AUX",),
+		KEYIDS["KEY_1"]: ("1BUTTON",),
+		KEYIDS["KEY_2"]: ("2BUTTON",),
+		KEYIDS["KEY_3"]: ("3BUTTON",),
+		KEYIDS["KEY_4"]: ("4BUTTON",),
+		KEYIDS["KEY_5"]: ("5BUTTON",),
+		KEYIDS["KEY_6"]: ("6BUTTON",),
+		KEYIDS["KEY_7"]: ("7BUTTON",),
+		KEYIDS["KEY_8"]: ("8BUTTON",),
+		KEYIDS["KEY_9"]: ("9BUTTON",),
+		KEYIDS["KEY_BACK"]: ("BACK",),
+		KEYIDS["KEY_0"]: ("0BUTTON",),
+		KEYIDS["KEY_HELP"]: ("INFO",),
+		KEYIDS["KEY_LANGUAGE"]: ("AUDIO",),
+		KEYIDS["KEY_SUBTITLE"]: ("SUBTITLE",),
+		KEYIDS["KEY_TEXT"]: ("TEXT",),
+		KEYIDS["KEY_MENU"]: ("MENU",),
+		KEYIDS["KEY_EPG"]: ("GUIDE",),
+		KEYIDS["KEY_UP"]: ("UP/P+",),
+		KEYIDS["KEY_LEFT"]: ("LEFT/V-",),
+		KEYIDS["KEY_OK"]: ("OK/LIST",),
+		KEYIDS["KEY_RIGHT"]: ("RIGHT/V+",),
+		KEYIDS["KEY_DOWN"]: ("DOWN/P-",),
+		KEYIDS["KEY_HOME"]: ("EXIT",),
+		KEYIDS["KEY_FAVORITES"]: ("FAV",),
+		KEYIDS["KEY_RED"]: ("RED",),
+		KEYIDS["KEY_GREEN"]: ("GREEN",),
+		KEYIDS["KEY_YELLOW"]: ("YELLOW",),
+		KEYIDS["KEY_BLUE"]: ("BLUE",),
+		KEYIDS["KEY_REWIND"]: ("REWIND",),
+		KEYIDS["KEY_PAUSE"]: ("PAUSE",),
+		KEYIDS["KEY_PLAY"]: ("PLAY",),
+		KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",),
+		KEYIDS["KEY_RECORD"]: ("RECORD",),
+		KEYIDS["KEY_STOP"]: ("STOP",),
+		KEYIDS["KEY_SLOW"]: ("SLOWMOTION",),
+		KEYIDS["KEY_LIST"]: ("RECORDLIST" ,),
+		KEYIDS["KEY_SAT"]: ("SAT",),
+		KEYIDS["KEY_PREVIOUS"]: ("STEPBACK",),
+		KEYIDS["KEY_NEXT"]: ("STEPFORWARD",),
+		KEYIDS["KEY_OPTION"]: ("MARK",),
+		KEYIDS["KEY_TV2"]: ("TV/RADIO",),
+		KEYIDS["KEY_ARCHIVE"]: ("USB",),
+		KEYIDS["KEY_TIME"]: ("TIMER",),
+		# Frontpanel keys
+		KEYIDS["KEY_PAGEDOWN"]: ("CHANNELDOWN",),
+		KEYIDS["KEY_PAGEUP"]: ("CHANNELUP",),
+		KEYIDS["KEY_VOLUMEDOWN"]: ("VOLUMEDOWN",),
+		KEYIDS["KEY_VOLUMEUP"]: ("VOLUMEUP",),
+		KEYIDS["KEY_POWER"]: ("STANDBY",),
+	},
+	{  # for remotes with ID 6 (Kathrein UFS912)
+		KEYIDS["KEY_MEDIA"]: ("MEDIA",),
+		KEYIDS["KEY_ARCHIVE"]: ("ARCHIVE",),
+		KEYIDS["KEY_MENU"]: ("MENU",),
+		KEYIDS["KEY_RED"]: ("RED",),
+		KEYIDS["KEY_GREEN"]: ("GREEN",),
+		KEYIDS["KEY_YELLOW"]: ("YELLOW",),
+		KEYIDS["KEY_BLUE"]: ("BLUE",),
+		KEYIDS["KEY_HOME"]: ("EXIT",),
+		KEYIDS["KEY_TEXT"]: ("TEXT",),
+		KEYIDS["KEY_EPG"]: ("EPG",),
+		KEYIDS["KEY_REWIND"]: ("REWIND",),
+		KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",),
+		KEYIDS["KEY_PLAY"]: ("PLAY",),
+		KEYIDS["KEY_PAUSE"]: ("PAUSE",),
+		KEYIDS["KEY_RECORD"]: ("RECORD",),
+		KEYIDS["KEY_STOP"]: ("STOP",),
+		KEYIDS["KEY_POWER"]: ("STANDBY",),
+		KEYIDS["KEY_MUTE"]: ("MUTE",),
+		KEYIDS["KEY_PAGEUP"]: ("CHANNELUP",),
+		KEYIDS["KEY_PAGEDOWN"]: ("CHANNELDOWN",),
+		KEYIDS["KEY_VOLUMEUP"]: ("VOLUMEUP",),
+		KEYIDS["KEY_VOLUMEDOWN"]: ("VOLUMEDOWN",),
+		KEYIDS["KEY_HELP"]: ("HELP",),
+		KEYIDS["KEY_INFO"]: ("INFO",),
+		KEYIDS["KEY_OK"]: ("OK",),
+		KEYIDS["KEY_UP"]: ("UP",),
+		KEYIDS["KEY_RIGHT"]: ("RIGHT",),
+		KEYIDS["KEY_DOWN"]: ("DOWN",),
+		KEYIDS["KEY_LEFT"]: ("LEFT",),
+		KEYIDS["KEY_0"]: ("0BUTTON",),
+		KEYIDS["KEY_1"]: ("1BUTTON",),
+		KEYIDS["KEY_2"]: ("2BUTTON",),
+		KEYIDS["KEY_3"]: ("3BUTTON",),
+		KEYIDS["KEY_4"]: ("4BUTTON",),
+		KEYIDS["KEY_5"]: ("5BUTTON",),
+		KEYIDS["KEY_6"]: ("6BUTTON",),
+		KEYIDS["KEY_7"]: ("7BUTTON",),
+		KEYIDS["KEY_8"]: ("8BUTTON",),
+		KEYIDS["KEY_9"]: ("9BUTTON",),
+		# Front panel keys
+		KEYIDS["KEY_MEDIA"]: ("FP_MEDIA",),
+		KEYIDS["KEY_POWER"]: ("FP_ON_OFF",),
+		KEYIDS["KEY_DOWN"]: ("FP_MINUS",),
+		KEYIDS["KEY_UP"]: ("FP_PLUS",),
+		KEYIDS["KEY_OK"]: ("FP_TV_R",),
 	},
 ]
 
@@ -246,10 +366,24 @@
 		idx = config.misc.rcused.value
 	else:
 		rctype = config.plugins.remotecontroltype.rctype.value
-		if rctype == 14:	# XP1000
+		print "[Keybindings.py] RC type is:", rctype
+		rctype = 11
+		if rctype == 10:	# Spark
 			idx = 3
-		elif rctype == 18:	# F1
+		elif rctype == 7:	# Fortis FS9000/FS9200/HS8200
 			idx = 4
+		elif rctype == 8:	# Fortis HS9510/HS7810A/HS7819
+			idx = 4
+		elif rctype == 9:	# Fortis HS7110/HS7119
+			idx = 4
+		elif rctype == 11:	# Topfield TF77X0HDPVR
+			idx = 5
+#		elif rctype == 12:	# Kathrein UFS912
+#			idx = 6
+#		elif rctype == 13:	# Cuberevo
+#			idx = 7
+#		elif rctype == 14:	# ...
+#			idx = 8
 		else:
 			idx = 2
 	if key in keyDescriptions[idx]:
--- enigma2-nightly.org/lib/python/Screens/InputDeviceSetup.py.org
+++ enigma2-nightly/lib/python/Screens/InputDeviceSetup.py
@@ -296,39 +296,21 @@
 class RemoteControlType(Screen, ConfigListScreen):
 	rcList = [
 			("0", _("Default")),
-			("11", _("et9200/9500/6500")),
-			("13", _("et4000")),
-			("7", _("et5000/6000")),
-			("9", _("et8000/et10000")),
 			("4", _("DMM normal")),
-			("5", _("et9000/et9100")),
 			("6", _("DMM advanced")),
-			("14", _("xp1000")),
-			("8", _("VU+")),
-			("18", _("F1/F3")),
-			("16", _("HD1100/et7x00/et8500")),
-			("19", _("HD2400"))
+			("7", _("fs9000")),
+			("8", _("hs9510")),
+			("9", _("hs7110")),
+			("10", _("spark")),
+			("11", _("tf7700")),
 		]
 
 	defaultRcList = [
-			("et4000", 13),
-			("et5000", 7),
-			("et6000", 7),
-			("et6500", 11),
-			("et8000", 9),
-			("et9000", 5),
-			("et9100", 5),
-			("et9200", 11),
-			("et9500", 11),
-			("et10000", 9),
-			("formuler1", 18),
-			("formuler3", 18),
-			("xp1000", 14),
-			("hd1100", 16),
-			("hd2400", 19),
-			("et7000", 16),
-			("et7500", 16),
-			("et8500", 16)
+			("fs9000", 7),
+			("hs9510", 8),
+			("hs7110", 9),
+			("spark", 10),
+			("tf7700", 11),
 		]
 
 	def __init__(self, session):
--- enigma2-nightly.org/data/rc_models/rc_models.cfg	2014-04-03 08:38:53.154073324 +0200
+++ enigma2-nightly/data/rc_models/rc_models.cfg	2014-05-28 20:18:22.000000000 +0200
@@ -19,34 +19,14 @@
 #
 # When no entry matches at all, 'default' will be used as remoteName
 #
-et10000  et8000
-et10000.4   dmm
-et10000.6   dmmadv
-et4000   et4x00
-et6500   et6x00
-et6500.4   dmm
-et6500.6   dmmadv
-et8000   et8000
-et8000.4   dmm
-et8000.6   dmmadv
-et9000   et9x00
-et9000.4   dmm
-et9000.6   dmmadv
-et9000.9  et9500
-et9200   et9x00
-et9500   et9500
-vuduo2   vuduo2
-vuduo    vu
-vusolo2  vu
-vusolo   vu
-vuultimo vuultimo
-vuuno    vu
-xp1000   xp1000
-vusolose vu
-vuzero   vu
-formuler1   formuler1
-et7000 et7x00
-et7500 et7x00
-et8500 et8000
-hd2400	hd2400
-hd1100	hd1100
+hdbox       fs9000
+octagon1008 hs9510
+atevio7500  fs9000
+hs7110      hs7110
+hs7119      hs7110
+hs7810a     hs9510
+hs7819      hs9510
+spark       spark
+spark7162   spark
+tf7700hdpvr tf7700
+
--- enigma2-nightly.org/data/rc_models/Makefile.am
+++ enigma2-nightly/data/rc_models/Makefile.am
@@ -2,9 +2,10 @@ 
 
 dist_install_DATA = \
 	rc_models.cfg \
-	dmmadv.png  et4x00.png  et8000.png  et9x00.png  vu.png        xp1000.png \
+	dmmadv.png \
+	fs9000.png  hs7110.png  hs9510.png  spark.png tf7700.png \
-	dmm.png     et6x00.png  et9500.png  vuduo2.png  vuultimo.png  et7x00.png    formuler1.png \
+	dmm.png \
-	hd2400.png  hd1100.png \
+	fs9000.xml  hs7110.xml  hs9510.xml  spark.xml tf7700.xml \
 	dmmadv.xml  et4x00.xml  et8000.xml  et9x00.xml  vuultimo.xml  xp1000.xml \
 	dmm.xml     et6x00.xml  et9500.xml  et7x00.xml  vuduo2.xml    vu.xml        formuler1.xml \
  	hd2400.xml  hd1100.xml
--- enigma2-nightly.org/data/rc_models/fs9000.png	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/fs9000.png	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,2 @@
+
+
--- enigma2-nightly.org/data/rc_models/fs9000/rcpositions.xml	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/fs9000.xml	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,57 @@
+<rcs>
+	<!-- Fortis FS9000/FS9200/HS8200-->
+	<rc id="2">
+		<button name="MUTE"        pos="86,18" />	<!-- KEY_MUTE-->
+		<button name="STANDBY"     pos="25,44" />	<!-- KEY_POWER-->
+		<button name="VFORMAT"     pos="63,44" />	<!-- KEY_ANGLE-->
+		<button name="RESOLUTION"  pos="86,44" />	<!-- KEY_SCREEN-->
+		<button name="1BUTTON"     pos="25,71" />	<!-- KEY_1-->
+		<button name="2BUTTON"     pos="55,71" />	<!-- KEY_2-->
+		<button name="3BUTTON"     pos="86,71" />	<!-- KEY_3-->
+		<button name="4BUTTON"     pos="25,96" />	<!-- KEY_4-->
+		<button name="5BUTTON"     pos="55,96" />	<!-- KEY_5-->
+		<button name="6BUTTON"     pos="86,96" />	<!-- KEY_6-->
+		<button name="7BUTTON"     pos="25,120" />	<!-- KEY_7-->
+		<button name="8BUTTON"     pos="55,120" />	<!-- KEY_8-->
+		<button name="9BUTTON"     pos="86,120" />	<!-- KEY_9-->
+		<button name="INFO"        pos="25,145" />	<!-- KEY_HELP THIS IS WRONG SHOULD BE KEY_INFO -->
+		<button name="0BUTTON"     pos="55,145" />	<!-- KEY_0-->
+		<button name="RECALL"      pos="86,145" />	<!-- KEY_MEMO-->
+		<button name="VOLUMEUP"    pos="27,172" />	<!-- KEY_VOLUMEUP-->
+		<button name="MENU"        pos="55,172" />	<!-- KEY_MENU-->
+		<button name="CHANNELUP"   pos="83,172" />	<!-- KEY_PAGEUP-->
+		<button name="VOLUMEDOWN"  pos="27,198" />	<!-- KEY_VOLUMEDOWN-->
+		<button name="EXIT"        pos="55,198" />	<!-- KEY_HOME-->
+		<button name="CHANNELDOWN" pos="83,198" />	<!-- KEY_PAGEDOWN-->
+		<button name="UP"          pos="55,228" />	<!-- KEY_UP-->
+		<button name="LEFT"        pos="27,247" />	<!-- KEY_LEFT-->
+		<button name="OK"          pos="55,247" />	<!-- KEY_OK-->
+		<button name="RIGHT"       pos="83,247" />	<!-- KEY_RIGHT-->
+		<button name="DOWN"        pos="55,265" />	<!-- KEY_DOWN-->
+		<button name="UPUP"        pos="25,292" />	<!-- KEY_TEEN-->
+		<button name="EPG"         pos="55,292" />	<!-- KEY_EPG-->
+		<button name="DOWNDOWN"    pos="50,292" />	<!-- KEY_TWEN-->
+		<button name="REWIND"      pos="25,314" />	<!-- KEY_REWIND-->
+		<button name="PLAY"        pos="55,314" />	<!-- KEY_PLAY-->
+		<button name="FASTFORWARD" pos="86,314" />	<!-- KEY_FASTFORWARD-->
+		<button name="LAST"        pos="25,337" />	<!-- KEY_PREVIOUS--> 
+		<button name="RECORD"      pos="55,337" />	<!-- KEY_RECORD-->
+		<button name="NEXT"        pos="86,337" />	<!-- KEY_NEXT-->
+		<button name="OPEN"        pos="21,358" />	<!-- KEY_DIRECTORY-->
+		<button name="PAUSE"       pos="44,360" />	<!-- KEY_PAUSE-->
+		<button name="STOP"        pos="67,360" />	<!-- KEY_STOP-->
+		<button name="CHECK"       pos="90,358" />	<!-- KEY_SELECT-->
+		<button name="RED"         pos="21,374" />	<!-- KEY_RED-->
+		<button name="GREEN"       pos="44,376" />	<!-- KEY_GREEN-->
+		<button name="YELLOW"      pos="67,376" />	<!-- KEY_YELLOW-->
+		<button name="BLUE"        pos="90,374" />	<!-- KEY_BLUE-->
+		<button name="PIP"         pos="21,391" />	<!-- KEY_OPTION-->
+		<button name="SWAP"        pos="44,393" />	<!-- KEY_GOTO-->
+		<button name="PIP_SEL"     pos="67,393" />	<!-- KEY_PIPSEL-->
+		<button name="SLEEP"       pos="90,391" />	<!-- KEY_SLOW-->
+		<button name="FAV"         pos="21,407" />	<!-- KEY_FAVORITES-->
+		<button name="TVRADIO"     pos="44,409" />	<!-- KEY_MODE-->
+		<button name="LIST"        pos="67,409" />	<!-- KEY_SUBTITLE-->
+		<button name="TEXT"        pos="90,407" />	<!-- KEY_TEXT-->
+	</rc>
+</rcs>
--- enigma2-nightly.org/data/rc_models/hs9510/rc.png	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/hs9510.png	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,2 @@
+
+
diff -Nur --exclude=.git --exclude=tuxtxt enigma2-nightly.org/data/rc_models/hs9510/rcpositions.xml enigma2-nightly.patched/data/rc_models/hs9510/rcpositions.xml
--- enigma2-nightly.org/data/rc_models/hs9510/rcpositions.xml	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/hs9510.xml	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,56 @@
+<rcs>
+	<!-- Fortis HS9510/HS7810A/HS7819-->
+	<rc id="2">
+		<button name="MUTE"        pos="88,34" />	<!-- KEY_MUTE-->
+		<button name="STANDBY"     pos="36,34" />	<!-- KEY_POWER-->
+		<button name="VFORMAT"     pos="29,64" />	<!-- KEY_ANGLE-->
+		<button name="RESOLUTION"  pos="90,64" />	<!-- KEY_SCREEN-->
+		<button name="1BUTTON"     pos="32,108" />	<!-- KEY_1-->
+		<button name="2BUTTON"     pos="61,108" />	<!-- KEY_2-->
+		<button name="3BUTTON"     pos="88,108" />	<!-- KEY_3-->
+		<button name="4BUTTON"     pos="32,125" />	<!-- KEY_4-->
+		<button name="5BUTTON"     pos="61,125" />	<!-- KEY_5-->
+		<button name="6BUTTON"     pos="88,125" />	<!-- KEY_6-->
+		<button name="7BUTTON"     pos="32,142" />	<!-- KEY_7-->
+		<button name="8BUTTON"     pos="61,142" />	<!-- KEY_8-->
+		<button name="9BUTTON"     pos="88,142" />	<!-- KEY_9-->
+		<button name="INFO"        pos="29,179" />	<!-- KEY_HELP THIS IS WRONG SHOULD BE KEY_INFO -->
+		<button name="0BUTTON"     pos="61,159" />	<!-- KEY_0-->
+		<button name="RECALL"      pos="91,179" />	<!-- KEY_MEMO-->
+		<button name="VOLUMEUP"    pos="35,262" />	<!-- KEY_VOLUMEUP-->
+		<button name="MENU"        pos="29,241" />	<!-- KEY_MENU-->
+		<button name="CHANNELUP"   pos="84,262" />	<!-- KEY_PAGEUP-->
+		<button name="VOLUMEDOWN"  pos="35,283" />	<!-- KEY_VOLUMEDOWN-->
+		<button name="EXIT"        pos="91,241" />	<!-- KEY_HOME-->
+		<button name="CHANNELDOWN" pos="84,283" />	<!-- KEY_PAGEDOWN-->
+		<button name="UP"          pos="60,185" />	<!-- KEY_UP-->
+		<button name="LEFT"        pos="34,210" />	<!-- KEY_LEFT-->
+		<button name="OK"          pos="60,210" />	<!-- KEY_OK-->
+		<button name="RIGHT"       pos="85,210" />	<!-- KEY_RIGHT-->
+		<button name="DOWN"        pos="60,235" />	<!-- KEY_DOWN-->
+		<button name="UPUP"        pos="30,86" />	<!-- KEY_TEEN-->
+		<button name="EPG"         pos="60,273" />	<!-- KEY_EPG-->
+		<button name="DOWNDOWN"    pos="50,86" />	<!-- KEY_TWEN-->
+		<button name="REWIND"      pos="31,332" />	<!-- KEY_REWIND-->
+		<button name="PLAY"        pos="50,332" />	<!-- KEY_PLAY-->
+		<button name="FASTFORWARD" pos="88,332" />	<!-- KEY_FASTFORWARD-->
+		<button name="LAST"        pos="31,350" />	<!-- KEY_PREVIOUS--> 
+		<button name="RECORD"      pos="69,350" />	<!-- KEY_RECORD-->
+		<button name="NEXT"        pos="88,350" />	<!-- KEY_NEXT-->
+		<button name="OPEN"        pos="88,159" />	<!-- KEY_DIRECTORY-->
+		<button name="PAUSE"       pos="69,332" />	<!-- KEY_PAUSE-->
+		<button name="STOP"        pos="50,350" />	<!-- KEY_STOP-->
+		<button name="CHECK"       pos="31,368" />	<!-- KEY_SELECT-->
+		<button name="RED"         pos="27,305" />	<!-- KEY_RED-->
+		<button name="GREEN"       pos="49,305" />	<!-- KEY_GREEN-->
+		<button name="YELLOW"      pos="70,305" />	<!-- KEY_YELLOW-->
+		<button name="BLUE"        pos="91,305" />	<!-- KEY_BLUE-->
+		<button name="PIP"         pos="50,368" />	<!-- KEY_OPTION-->
+		<button name="SWAP"        pos="69,368" />	<!-- KEY_GOTO-->
+		<button name="PIP_SEL"     pos="88,368" />	<!-- KEY_PIPSEL-->
+		<button name="SLEEP"       pos="50,64" />	<!-- KEY_SLOW-->
+		<button name="FAV"         pos="32,159" />	<!-- KEY_FAVORITES-->
+		<button name="TVRADIO"     pos="70,64" />	<!-- KEY_MODE-->
+		<button name="LIST"        pos="70,86" />	<!-- KEY_SUBTITLE-->
+		<button name="TEXT"        pos="90,86" />	<!-- KEY_TEXT-->
+	</rc>
+</rcs>
diff -Nur --exclude=.git --exclude=tuxtxt enigma2-nightly.org/data/rc_models/hs7110/rc.png enigma2-nightly.patched/data/rc_models/hs7110/rc.png
--- enigma2-nightly.org/data/rc_models/hs7110/rc.png	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/hs7110.png	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,2 @@
+
+
diff -Nur --exclude=.git --exclude=tuxtxt enigma2-nightly.org/data/rc_models/hs7110/rcpositions.xml enigma2-nightly.patched/data/rc_models/hs7110/rcpositions.xml
--- enigma2-nightly.org/data/rc_models/hs7110/rcpositions.xml	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/hs7110.xml	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,57 @@
+<rcs>
+	<!-- Fortis HS7110//HS7119-->
+	<rc id="2">
+		<button name="MUTE"        pos="91,29" />	<!-- KEY_MUTE-->
+		<button name="STANDBY"     pos="33,29" />	<!-- KEY_POWER-->
+		<!--button name="VFORMAT"     pos="63,44" /-->	<!-- KEY_ANGLE-->
+		<button name="RESOLUTION"  pos="89,52" />	<!-- KEY_SCREEN-->
+		<button name="1BUTTON"     pos="32,53" />	<!-- KEY_1-->
+		<button name="2BUTTON"     pos="61,53" />	<!-- KEY_2-->
+		<button name="3BUTTON"     pos="89,53" />	<!-- KEY_3-->
+		<button name="4BUTTON"     pos="31,76" />	<!-- KEY_4-->
+		<button name="5BUTTON"     pos="61,76" />	<!-- KEY_5-->
+		<button name="6BUTTON"     pos="91,76" />	<!-- KEY_6-->
+		<button name="7BUTTON"     pos="30,98" />	<!-- KEY_7-->
+		<button name="8BUTTON"     pos="61,98" />	<!-- KEY_8-->
+		<button name="9BUTTON"     pos="92,98" />	<!-- KEY_9-->
+		<button name="INFO"        pos="99,376" />	<!-- KEY_HELP THIS IS WRONG SHOULD BE KEY_INFO -->
+		<button name="0BUTTON"     pos="61,120" />	<!-- KEY_0-->
+		<button name="RECALL"      pos="97,236" />	<!-- KEY_MEMO-->
+		<button name="VOLUMEUP"    pos="26,261" />	<!-- KEY_VOLUMEUP-->
+		<button name="MENU"        pos="27,208" />	<!-- KEY_MENU-->
+		<button name="CHANNELUP"   pos="98,261" />	<!-- KEY_PAGEUP-->
+		<button name="VOLUMEDOWN"  pos="28,285" />	<!-- KEY_VOLUMEDOWN-->
+		<button name="EXIT"        pos="97,208" />	<!-- KEY_HOME-->
+		<button name="CHANNELDOWN" pos="98,285" />	<!-- KEY_PAGEDOWN-->
+		<button name="UP"          pos="62,141" />	<!-- KEY_UP-->
+		<button name="LEFT"        pos="36,168" />	<!-- KEY_LEFT-->
+		<button name="OK"          pos="62,168" />	<!-- KEY_OK-->
+		<button name="RIGHT"       pos="88,168" />	<!-- KEY_RIGHT-->
+		<button name="DOWN"        pos="62,194" />	<!-- KEY_DOWN-->
+		<button name="UPUP"        pos="28,376" />	<!-- KEY_TEEN-->
+		<button name="EPG"         pos="63,236" />	<!-- KEY_EPG-->
+		<button name="DOWNDOWN"    pos="51,376" />	<!-- KEY_TWEN-->
+		<button name="REWIND"      pos="29,310" />	<!-- KEY_REWIND-->
+		<button name="PLAY"        pos="63,259" />	<!-- KEY_PLAY-->
+		<button name="FASTFORWARD" pos="98,310" />	<!-- KEY_FASTFORWARD-->
+		<button name="LAST"        pos="29,330" />	<!-- KEY_PREVIOUS--> 
+		<button name="RECORD"      pos="64,330" />	<!-- KEY_RECORD-->
+		<button name="NEXT"        pos="98,330" />	<!-- KEY_NEXT-->
+		<button name="OPEN"        pos="63,217" />	<!-- KEY_DIRECTORY-->
+		<button name="PAUSE"       pos="63,287" />	<!-- KEY_PAUSE-->
+		<button name="STOP"        pos="63,310" />	<!-- KEY_STOP-->
+		<!--button name="CHECK"       pos="90,358" /-->	<!-- KEY_SELECT-->
+		<button name="RED"         pos="27,351" />	<!-- KEY_RED-->
+		<button name="GREEN"       pos="51,351" />	<!-- KEY_GREEN-->
+		<button name="YELLOW"      pos="75,351" />	<!-- KEY_YELLOW-->
+		<button name="BLUE"        pos="99,351" />	<!-- KEY_BLUE-->
+		<button name="PIP"         pos="39,400" />	<!-- KEY_OPTION-->
+		<button name="SWAP"        pos="63,400" />	<!-- KEY_GOTO-->
+		<button name="PIP_SEL"     pos="88,400" />	<!-- KEY_PIPSEL-->
+		<!--button name="SLEEP"       pos="90,391" /-->	<!-- KEY_SLOW-->
+		<button name="FAV"         pos="27,236" />	<!-- KEY_FAVORITES-->
+		<button name="TVRADIO"     pos="76,376" />	<!-- KEY_MODE-->
+		<button name="LIST"        pos="31,120" />	<!-- KEY_SUBTITLE-->
+		<button name="TEXT"        pos="94,120" />	<!-- KEY_TEXT-->
+	</rc>
+</rcs>
diff -Nur --exclude=.git --exclude=tuxtxt enigma2-nightly.org/data/rc_models/spark/rc.png enigma2-nightly.patched/data/rc_models/spark/rc.png
--- enigma2-nightly.org/data/rc_models/spark/rc.png	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/spark.png	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,2 @@
+
+
diff -Nur --exclude=.git --exclude=tuxtxt enigma2-nightly.org/data/rc_models/spark/rcpositions.xml enigma2-nightly.patched/data/rc_models/spark/rcpositions.xml
--- enigma2-nightly.org/data/rc_models/spark/rcpositions.xml	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/spark.xml	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,60 @@
+<rcs>
+	<!-- Spark RC12-->
+	<rc id="2">
+		<button name="POWER" pos="53,19" />	<!-- KEY_POWER-->
+		<button name="V.FORMAT" pos="22,36" />	<!-- KEY_V-->
+		<button name="TV/SAT" pos="45,36" />	<!-- KEY_AUX-->
+		<button name="TIME" pos="64,36" />	<!-- KEY_TIME-->
+		<button name="MUTE" pos="83,36" />	<!-- KEY_MUTE-->
+		<button name="0" pos="53,122" />	<!-- KEY_0-->
+		<button name="1" pos="25,56" />		<!-- KEY_1-->
+		<button name="2" pos="53,56" />		<!-- KEY_2-->
+		<button name="3" pos="81,56" />		<!-- KEY_3-->
+		<button name="4" pos="25,78" />		<!-- KEY_4-->
+		<button name="5" pos="53,78" />		<!-- KEY_5-->
+		<button name="6" pos="81,78" />		<!-- KEY_6-->
+		<button name="7" pos="25,100" />	<!-- KEY_7-->
+		<button name="8" pos="53,100" />	<!-- KEY_8-->
+		<button name="9" pos="81,100" />	<!-- KEY_9-->
+		<button name="INFO" pos="25,145" />	<!-- KEY_HELP THIS IS WRONG SHOULD BE KEY_INFO -->
+		<button name="TV/RADIO" pos="24,123" />	<!-- KEY_TV2-->
+		<button name="RECALL" pos="81,123" />	<!-- KEY_BACK-->
+		<button name="PAGE+" pos="27,146" />	<!-- KEY_PAGEUP-->
+		<button name="PAGE-" pos="27,174" />	<!-- KEY_PAGEDOWN-->
+		<button name="FIND" pos="53,147" />	<!-- KEY_FIND-->
+		<button name="FOLDER" pos="53,172" />	<!-- KEY_ARCHIVE-->
+		<button name="VOL+" pos="80,146" />	<!-- KEY_VOLUMEUP-->
+		<button name="VOL-" pos="80,174" />	<!-- KEY_VOLUMEDOWN-->
+		<button name="MENU" pos="24,196" />	<!-- KEY_MENU-->
+		<button name="INFO" pos="81,196" />	<!-- KEY_INFO-->
+		<button name="UP" pos="53,206" />	<!-- KEY_UP-->
+		<button name="DOWN" pos="53,250" />	<!-- KEY_DOWN-->
+		<button name="LEFT" pos="31,230" />	<!-- KEY_LEFT-->
+		<button name="RIGHT" pos="76,230" />	<!-- KEY_RIGHT-->
+		<button name="OK" pos="53,230" />	<!-- KEY_OK-->
+		<button name="EXIT" pos="101,265" />	<!-- KEY_HOME-->
+		<button name="EPG" pos="81,265" />	<!-- KEY_EPG-->
+		<button name="PORTAL" pos="53,280" />	<!-- KEY_OPEN -->
+		<button name="FAV" pos="23,286" />	<!-- KEY_FAVORITES-->
+		<button name="SAT" pos="81,286" />	<!-- KEY_SAT-->
+		<button name="RED" pos="23,305" />	<!-- KEY_RED-->
+		<button name="GREEN" pos="42,305" />	<!-- KEY_GREEN-->
+		<button name="YELLOW" pos="62,305" />	<!-- KEY_YELLOW-->
+		<button name="BLUE" pos="81,305" />	<!-- KEY_BLUE-->
+		<button name="RECORD" pos="23,324" />	<!-- KEY_RECORD-->
+		<button name="STOP" pos="42,324" />	<!-- KEY_STOP-->
+		<button name="PAUSE" pos="62,324" />	<!-- KEY_PAUSE-->
+		<button name="PLAY" pos="83,324" />	<!-- KEY_PLAY-->
+		<button name="ARROWLEFT" pos="23,341" />	<!-- KEY_PREVIOUS-->
+		<button name="ARROWRIGHT" pos="42,341" />	<!-- KEY_NEXT-->
+		<button name="REWIND" pos="62,341" />	<!-- KEY_REWIND-->
+		<button name="FASTFORWARD" pos="83,341" />	<!-- KEY_FASTFORWARD-->
+		<button name="FAST" pos="23,358" />	<!-- KEY_F-->
+		<button name="SLOW" pos="42,358" />	<!-- KEY_SLOW-->
+		<button name="PLAY_MODE" pos="62,358" />	<!-- KEY_P-->
+		<button name="USB" pos="83,358" />	<!-- KEY_CLOSE-->
+		<button name="Tms" pos="23,374" />	<!-- KEY_T-->
+		<button name="F1" pos="42,374" />	<!-- KEY_F1-->
+		<button name="F2" pos="62,374" />	<!-- KEY_F2-->
+		<button name="F3" pos="83,374" />	<!-- KEY_F3-->
+	</rc>
+</rcs>
diff -Nur --exclude=.git --exclude=tuxtxt enigma2-nightly.org/data/rc_models/tf7700/rc.png enigma2-nightly.patched/data/rc_models/tf7700/rc.png
--- enigma2-nightly.org/data/rc_models/tf7700/rc.png	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/tf7700.png	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,2 @@
+
+
diff -Nur --exclude=.git --exclude=tuxtxt enigma2-nightly.org/data/rc_models/tf7700/rcpositions.xml enigma2-nightly.patched/data/rc_models/tf7700/rcpositions.xml
--- enigma2-nightly.org/data/rc_models/tf7700/rcpositions.xml	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/data/rc_models/tf7700.xml	2013-12-18 20:00:00.000000000 +0100
@@ -0,0 +1,53 @@
+<rcs>
+	<!-- Topfield TD77X0HDPVR-->
+	<rc id="2">
+		<button name="STANDBY"     pos="31,35" />	<!-- KEY_POWER-->
+		<button name="MUTE"        pos="81,35" />	<!-- KEY_MUTE-->
+		<button name="V.FORMAT"    pos="27,61" />	<!-- KEY_ZOOM-->
+		<button name="A/R"         pos="57,61" />	<!-- KEY_SCREEN-->
+		<button name="AUX"         pos="87,61" />	<!-- KEY_AUX-->
+		<button name="1BUTTON"     pos="27,84" />	<!-- KEY_1-->
+		<button name="2BUTTON"     pos="57,84" />	<!-- KEY_2-->
+		<button name="3BUTTON"     pos="87,84" />	<!-- KEY_3-->
+		<button name="4BUTTON"     pos="27,104" />	<!-- KEY_4-->
+		<button name="5BUTTON"     pos="57,104" />	<!-- KEY_5-->
+		<button name="6BUTTON"     pos="87,104" />	<!-- KEY_6-->
+		<button name="7BUTTON"     pos="27,125" />	<!-- KEY_7-->
+		<button name="8BUTTON"     pos="57,125" />	<!-- KEY_8-->
+		<button name="9BUTTON"     pos="87,125" />	<!-- KEY_9-->
+		<button name="BACK"        pos="27,146" />	<!-- KEY_BACK-->
+		<button name="0BUTTON"     pos="57,146" />	<!-- KEY_0-->
+		<button name="INFO"        pos="87,146" />	<!-- KEY_HELP-->
+		<button name="AUDIO"       pos="27,167" />	<!-- KEY_LANGUAGE-->
+		<button name="SUBTITLE"    pos="57,167" />	<!-- KEY_SUBTITLE-->
+		<button name="TEXT"        pos="87,167" />	<!-- KEY_TEXT-->
+		<button name="MENU"        pos="29,197" />	<!-- KEY_MENU-->
+		<button name="GUIDE"       pos="83,197" />	<!-- KEY_EPG-->
+		<button name="UP/P+"       pos="57,208" />	<!-- KEY_UP-->
+		<button name="LEFT/V-"     pos="32,230" />	<!-- KEY_LEFT-->
+		<button name="OK/LIST"     pos="57,230" />	<!-- KEY_OK-->
+		<button name="RIGHT/V+"    pos="81,230" />	<!-- KEY_RIGHT-->
+		<button name="DOWN/P-"     pos="57,252" />	<!-- KEY_DOWN-->
+		<button name="EXIT"        pos="29,270" />	<!-- KEY_HOME-->
+		<button name="FAV"         pos="83,270" />	<!-- KEY_FAVORITES-->
+		<button name="RED"         pos="25,298" />	<!-- KEY_RED-->
+		<button name="GREEN"       pos="46,298" />	<!-- KEY_GREEN-->
+		<button name="YELLOW"      pos="67,298" />	<!-- KEY_YELLOW-->
+		<button name="BLUE"        pos="88,298" />	<!-- KEY_BLUE-->
+		<button name="REWIND"      pos="25,323" />	<!-- KEY_REWIND-->
+		<button name="PAUSE"       pos="46,323" />	<!-- KEY_PAUSE-->
+		<button name="PLAY"        pos="67,323" />	<!-- KEY_PLAY-->
+		<button name="FASTFORWARD" pos="88,323" />	<!-- KEY_FASTFORWARD-->
+		<button name="RECORD"      pos="25,346" />	<!-- KEY_RECORD-->
+		<button name="STOP"        pos="46,346" />	<!-- KEY_STOP-->
+		<button name="SLOWMOTION"  pos="67,346" />	<!-- KEY_SLOW-->
+		<button name="RECORDLIST"  pos="88,346" />	<!-- KEY_LIST-->
+		<button name="SAT"         pos="25,371" />	<!-- KEY_SAT-->
+		<button name="STEPBACK"    pos="46,371" />	<!-- KEY_PREVIOUS-->
+		<button name="STEPFORWARD" pos="67,371" />	<!-- KEY_NEXT-->
+		<button name="MARK"        pos="88,371" />	<!-- KEY_EPG-->
+		<button name="TV/RADIO"    pos="27,396" />	<!-- KEY_TV2-->
+		<button name="USB"         pos="57,396" />	<!-- KEY_ARCHIVE-->
+		<button name="TIMER"       pos="87,396" />	<!-- KEY_TIME-->
+	</rc>
+</rcs>
